lib/tests/streamlit/runtime/caching/cache_data_api_test.py

```diff
@@ -335,7 +335,9 @@ def foo(val):
 
         mock_os_remove.assert_not_called()
 
-        with patch("os.listdir", MagicMock(return_value=created_files_base_names)):
+        with patch(
+            "os.listdir", MagicMock(return_value=created_files_base_names)
+        ), patch("os.path.isdir", MagicMock(return_value=True)):
             # Clear foo's cache
             foo.clear()
```

Comment:
Would also be good to have a test that `listdir` is not called if `isdir` returns false

Reply:
Done!
I added tests for that in `local_disk_cache_storage_test.py`